src/lib/core/clock_lowres.c

```diff
@@ -0,0 +1,92 @@
+/*
+ * SPDX-License-Identifier: BSD-2-Clause
+ *
+ * Copyright 2010-2022, Tarantool AUTHORS, please see AUTHORS file.
+ */
+#include <signal.h>
+#include <string.h>
+#include <say.h>
+#include <pthread.h>
+#include "clock.h"
+#include "clock_lowres.h"
+
+#if defined(__APPLE__) || !defined(NDEBUG)
+/**
+ * A thread that initialized this module. Only owner thread
+ * is allowed to handle SIGALRM and use methods from this module.
+ */
+pthread_t owner;
+
+/** Check if current thread is owner. */
+bool
+clock_lowres_thread_is_owner(void)
+{
+	return pthread_self() == owner;
```

Comment:
You should use `pthread_equal` to compare pthread ids.

Reply:
fixed